.github/actions/setup-nix/action.yml

```diff
@@ -0,0 +1,66 @@
+name: setup-nix
+
+inputs:
+  system:
+    type: string
+    required: true
+  sandbox:
+    type: string
+    default: "true"
+
+runs:
+  using: composite
+  steps:
+    - name: reclaim space (linux)
+      if: runner.os == 'Linux'
+      uses: wimpysworld/nothing-but-nix@main
+      with:
+        hatchet-protocol: rampage
+
+    - name: reclaim space (darwin)
+      if: runner.os == 'macOS'
+      shell: bash
+      run: |
+        echo "::group::disk space (before)"
+        sudo df -h
+        echo "::endgroup::"
```

Comment:
If we have a composite action, we may not need explicit groups. How do they compare visually to splitting the groups into separate steps?